core/src/main/scala/org/apache/spark/deploy/worker/Worker.scala

```diff
@@ -466,23 +531,13 @@ private[worker] class Worker(
         case _ =>
           logDebug(s"Driver $driverId changed state to $state")
       }
-      master ! DriverStateChanged(driverId, state, exception)
+      sendToMaster(driverStageChanged)
```

Comment:
Rather, the change about the `driverStageChanged @` instead of recreating `DriverStateChanged(driverId, state, exception)`. It's not particularly important, but I was just curious if there was a reason.